test/functional/zmq_test.py

```diff
@@ -13,6 +13,25 @@
                                  hash256,
                                 )
 
+class ZMQSubscriber:
+    def __init__(self, socket, type):
```

Comment:
[type is a builtin](https://docs.python.org/3/library/functions.html#type). Perhaps `notification _type` is better?